### PR TITLE
Complete theme settings: toggle themes & apply system themes

### PR DESCRIPTION
**Pull request by kesesek**
Hi. I've implemented most of the dark mode CSS, but there might be something missing. Please review and let me know. If you need to make edits for dark mode, please do so in the global.scss file.

### CODE REVIEW

**Comment by ryanwooa** [comment]
image
These three area's color in dark mode need to update.

image
The styles change between switch button and click function on these two different colored picture need to be edit. They are not synchronous.

File: styles/scss/globals.scss [hunk 1]
```diff
@@ -20,6 +24,97 @@
...
+    background-color: none;
+    color: $secondaryColor;
+  }
+
+  #task_discription,
+  #task_label {
+    color: $grayColor;
+  }
+}
+
```
Comment (zli213) [inline styles/scss/globals.scss:117]: I think I understand why you chose to use IDs. If using classNames doesn't achieve the desired effect, please disregard my previous comments about IDs. However, if it's also feasible with classNames, I suggest trying to maintain consistency for uniformity.

File: styles/scss/globals.scss [hunk 2]
```diff
@@ -1,5 +1,9 @@
 // App.scss
 @use "./utils/" as *;
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+
```
Comment (zli213) [inline styles/scss/globals.scss:6]: May I ask where you have implemented Tailwind CSS?

**Review by zli213 — COMMENTED** [review]
Code modification suggestions can be found in detailed annotations.

File: app/api/auth/[...nextauth]/options.js [hunk 3]
```diff
@@ -142,6 +148,7 @@ export const options = {
         if (user[0]) {
           session.user.userId = user[0]._id.toString();
           session.user.username = user[0].username;
+          // session.user.theme = user[0].themes[0];
```
[the remark this answers is not in the document]
Comment (kesesek) [reply app/api/auth/[...nextauth]/options.js:151]: removed.

File: .env [hunk 4]
```diff
@@ -4,5 +4,8 @@ MONGO_URL = mongodb+srv://todo123:Z7xs799sOdztS5PU@tododata.ud04wyd.mongodb.net/
 # GOOGLE_ID=""
 # GOOGLE_Secret=""
 # NEXTAUTH_SECRET="Entert ramdom string here, and then run the command 'openssl rand -base64 32 ' in the terminal to generate a random string and replace it with the string here"
-
+GITHUB_ID=6abec741683c13078701
```
[the remark this answers is not in the document]
Comment (kesesek) [reply .env]: Sorry I've deleted it

File: styles/scss/globals.scss (hunk 1 above)
Comment (kesesek) [reply styles/scss/globals.scss:128]: I tried to use className but it didn't work, so maybe just keep it this way.

File: styles/scss/globals.scss (hunk 2 above)
Comment (kesesek) [reply styles/scss/globals.scss:6]: I tried to use it to before . Now it's deleted.

**Comment by kesesek** [comment]
> image The styles change between switch button and click function on these two different colored picture need to be edit. They are not synchronous.
> 
> And I cant switch back to account setting page when I in theme page
> 
> image Color for tag

It works normally on my device.Could you check it again please?  Edited tag colors.

**Comment by ryanwooa** [comment]
> > image The styles change between switch button and click function on these two different colored picture need to be edit. They are not synchronous.
> > And I cant switch back to account setting page when I in theme page
> > image Color for tag
> 
> It works normally on my device.Could you check it again please? Edited tag colors.

I mean, when I use pic to choose the dark mode, the switch is still on the left side ,which is the light mode side.
Maybe they should stay the same side?
image

**Comment by kesesek** [comment]
> > > image The styles change between switch button and click function on these two different colored picture need to be edit. They are not synchronous.
> > > And I cant switch back to account setting page when I in theme page
> > > image Color for tag
> > 
> > 
> > It works normally on my device.Could you check it again please? Edited tag colors.
> 
> I mean, when I use pic to choose the dark mode, the switch is still on the left side ,which is the light mode side. Maybe they should stay the same side? image

Oh I got it. This is because the switcher is used to sync themes with the user's system settings; when the user changes themes on their device, the website's theme will also change accordingly. Two icons are provided for manual toggling. Regardless of the user's device mode, they can freely toggle between themes.
I guess it's kind of confusing, maybe I should add more description beside icons? I'll edit the circle later.

**Comment by ryanwooa** [comment]
Some style bugs need to be fix
1.Hover background and font color
image

2. Number in the today Icon
image

3. font color in TaskEditor and border background color
image

4. style for complete button
image

5. color for button
image

6. Dark style for Search
image

**Comment by ryanwooa** [comment]
> > > > image The styles change between switch button and click function on these two different colored picture need to be edit. They are not synchronous.
> > > > And I cant switch back to account setting page when I in theme page
> > > > image Color for tag
> > > 
> > > 
> > > It works normally on my device.Could you check it again please? Edited tag colors.
> > 
> > 
> > I mean, when I use pic to choose the dark mode, the switch is still on the left side ,which is the light mode side. Maybe they should stay the same side? image
> 
> Oh I got it. This is because the switcher is used to sync themes with the user's system settings; when the user changes themes on their device, the website's theme will also change accordingly. Two icons are provided for manual toggling. Regardless of the user's device mode, they can freely toggle between themes. I guess it's kind of confusing, maybe I should add more description beside icons? I'll edit the circle later.

I think two way to switch the theme is great, just need to keep them synchronous.

**Comment by kesesek** [comment]
> > > > > image The styles change between switch button and click function on these two different colored picture need to be edit. They are not synchronous.
> > > > > And I cant switch back to account setting page when I in theme page
> > > > > image Color for tag
> > > > 
> > > > 
> > > > It works normally on my device.Could you check it again please? Edited tag colors.
> > > 
> > > 
> > > I mean, when I use pic to choose the dark mode, the switch is still on the left side ,which is the light mode side. Maybe they should stay the same side? image
> > 
> > 
> > Oh I got it. This is because the switcher is used to sync themes with the user's system settings; when the user changes themes on their device, the website's theme will also change accordingly. Two icons are provided for manual toggling. Regardless of the user's device mode, they can freely toggle between themes. I guess it's kind of confusing, maybe I should add more description beside icons? I'll edit the circle later.
> 
> I think two way to switch the theme is great, just need to keep them synchronous.

yes they are, you could open auto dark mode and try changing your device themes. I'll fix what you mentioned later.

**Comment by ryanwooa** [comment]
> Some style bugs need to be fix 1.Hover background and font color image
> 
> 2. Number in the today Icon
>    image
> 3. font color in TaskEditor and border background color
>    image
> 4. style for complete button
>    image
> 5. color for button
>    image
> 6. Dark style for Search
>    image

3 and 4 need be updated.

**Comment by ryanwooa** [comment]
These are solved, but we have new questions 
1. The sidebar is not displayed after refresh, why?
2. the height of setting? It should be fixed.
image

**Comment by ryanwooa** [comment]
4. the backgroud of buttons are missing
image

**Review by zli213 — APPROVED** [review]

File: components/pages/AppPages/Task/TaskHeaderLeft.js [hunk 5]
```diff
@@ -48,17 +48,17 @@ export default function TaskHeaderLeft({
               <Icon type="hashtag_small" />
             </span>
           )}
-          &nbsp;<span className={styles.tag_box2} style={{display: 'block'}}>{projectName}</span>
+          &nbsp;<span className={styles.tag_box2} style={{display: 'block'}} id="tag">{projectName}</span>
           {board && (
             <>
               &nbsp;/&nbsp;
-              <span className={styles.tag_box3}>{board}</span>
+              <span className={styles.tag_box3} id="tag">{board}</span>
```
Comment (jlmjkfd) [inline components/pages/AppPages/Task/TaskHeaderLeft.js:55]: Same id in different elements

File: components/pages/Settings/Subsetting/Delete.js [hunk 6]
```diff
@@ -52,13 +52,13 @@ const DeleteAccount = () => {
...
-          <button onClick={() => router.push("/application/setting/account")}>
-            <Icon type="left_arrow" />
+          <button onClick={() => router.push("/application/setting/account")} id="option_link">
+            <Icon type="left_arrow" id="icon"/>
           </button>
           <label>Modify your password</label>
         </span>
 
-        <button onClick={onDismiss}>
+        <button onClick={onDismiss} id="option_link">
```
Comment (jlmjkfd) [inline components/pages/Settings/Subsetting/Delete.js:61]: same id in different elements

File: components/pages/Settings/Subsetting/Email.js [hunk 7]
```diff
@@ -87,13 +87,13 @@ const SetEmail = () => {
...
         <span>
-          <button onClick={() => router.push("/application/setting/account")}>
+          <button onClick={() => router.push("/application/setting/account")} id="option_link">
             <Icon type="left_arrow" />
           </button>
           <label>Modify your email address</label>
         </span>
-        <button onClick={onDismiss}>
-          <Icon type="close" />
+        <button onClick={onDismiss} id="option_link">
```
Comment (jlmjkfd) [inline components/pages/Settings/Subsetting/Email.js:95]: same id in line 90

File: components/pages/Settings/Subsetting/Email.js [hunk 8]
```diff
@@ -123,7 +123,7 @@ const SetEmail = () => {
         </div>
 
         <span>
-          <p className={backendMessage ? styles.visible : styles.hidden}>
+          <p className={backendMessage ? styles.visible : styles.hidden} id="content_holder">
```
Comment (jlmjkfd) [inline components/pages/Settings/Subsetting/Email.js:126]: same id in line 114

File: components/pages/Settings/Subsetting/Password.js [hunk 9]
```diff
@@ -95,14 +95,14 @@ const SetPassword = () => {
...
-          <button onClick={() => router.push("/application/setting/account")}>
+          <button onClick={() => router.push("/application/setting/account")} id="option_link">
             <Icon type="left_arrow" />
           </button>
           <label>Modify your password</label>
         </span>
 
-        <button onClick={onDismiss}>
-          <Icon type="close" />
+        <button onClick={onDismiss} id="option_link">
```
Comment (jlmjkfd) [inline components/pages/Settings/Subsetting/Password.js:104]: same id in line 98

File: components/pages/Settings/Subsetting/Password.js [hunk 10]
```diff
@@ -133,14 +133,14 @@ const SetPassword = () => {
...
-            <p className={samePassword ? styles.hidden : styles.visible}>
+            <p className={samePassword ? styles.hidden : styles.visible} id="content_holder">
               Passwords do not match
             </p>
           </div>
         </div>
 
         <span>
-          <p className={backendMessage ? styles.visible : styles.hidden}>
+          <p className={backendMessage ? styles.visible : styles.hidden} id="content_holder">
```
Comment (jlmjkfd) [inline components/pages/Settings/Subsetting/Password.js:143]: same id in line 136

**Review by jlmjkfd — COMMENTED** [review]
You may ignore the same id problem if you can ensure only one element with the id will be rendered.

**Comment by ryanwooa** [comment]
> 4. the backgroud of buttons are missing
>    image
> 5. the border radium are not same in different themes.
>    image

question 2 and 4 still need be updated

**Review by ryanwooa — APPROVED** [review]

**Pull request merged**